experimenter/experimenter/nimbus_ui_new/forms.py

```diff
@@ -389,6 +393,258 @@ def get_changelog_message(self):
+            "localizations",
+            "prevent_pref_conflicts",
+            "warn_feature_schema",
+        )
+        widgets = {
+            "is_rollout": forms.CheckboxInput(attrs={"class": "form-check-input"}),
+            "warn_feature_schema": forms.CheckboxInput(
+                attrs={"class": "form-check-input"}
+            ),
+            "equal_branch_ratio": forms.CheckboxInput(
+                attrs={"class": "form-check-input"}
+            ),
+            "prevent_pref_conflicts": forms.CheckboxInput(
+                attrs={"class": "form-check-input"}
+            ),
+            "is_localized": forms.CheckboxInput(attrs={"class": "form-check-input"}),
+            "localizations": forms.Textarea(attrs={"class": "form-control", "rows": 5}),
+        }
+
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+
+        self.NimbusNimbusBranchFormSet = inlineformset_factory(
+            NimbusExperiment,
+            NimbusBranch,
+            form=NimbusBranchForm,
+            extra=0,
+        )
+        self.branches = self.NimbusNimbusBranchFormSet(
+            data=self.data or None,
+            instance=self.instance,
+        )
+
+        self.fields["feature_configs"].queryset = NimbusFeatureConfig.objects.filter(
+            application=self.instance.application
+        ).order_by("slug")
+
+        update_on_change_attrs = {
+            "hx-post": reverse(
+                "nimbus-new-partial-update-branches", kwargs={"slug": self.instance.slug}
+            ),
+            "hx-trigger": "change",
+            "hx-select": "#branches",
+            "hx-target": "#branches",
+        }
+        self.fields["is_rollout"].widget.attrs.update(update_on_change_attrs)
+        self.fields["feature_configs"].widget.attrs.update(update_on_change_attrs)
+        self.fields["equal_branch_ratio"].widget.attrs.update(update_on_change_attrs)
+        self.fields["is_localized"].widget.attrs.update(
+            {
+                **update_on_change_attrs,
+                "hx-select": "#localization",
+                "hx-target": "#localization",
+            }
+        )
+
+    @property
+    def errors(self):
+        errors = super().errors
+        if any(self.branches.errors):
+            errors["branches"] = self.branches.errors
+        return errors
+
+    def save(self, *args, **kwargs):
+        experiment = super().save(*args, **kwargs)
+        self.branches.save()
+
+        if experiment.is_rollout:
+            branches = experiment.branches.all()
+
+            if experiment.reference_branch:
+                branches = branches.exclude(id=experiment.reference_branch.id)
+
+            branches.delete()
+
+        saved_experiment_feature_configs = set(experiment.feature_configs.all())
+        saved_branch_feature_configs = {
+            fv.feature_config
+            for b in experiment.branches.all()
+            for fv in b.feature_values.all()
```

Comment:
This is an n*m query. We should turn this into two explicit queries.

Reply:
Sorry I meant its n+1 instead of n*m:

- `experiment.branches.all()` is one query
- `b.feature_values.all()`

we should do two explicit queries:

```python
bs = list(experiment.branches.all())
fvs = FeatureValue.objects.filter(experiment=experiment, branch__in=bs)
```

Reply:
even better!